File: app/src/main/java/com/mmutert/freshfreezer/data/converters/ItemAndNotifications.java
package com.mmutert.freshfreezer.data.converters;


import androidx.room.Embedded;
import androidx.room.Relation;

import com.mmutert.freshfreezer.data.FrozenItem;
import com.mmutert.freshfreezer.data.ItemNotification;

import java.util.List;


public class ItemAndNotifications {

    @Embedded
    public FrozenItem item;

    @Relation(parentColumn = "id", entityColumn = "item_id")
    public List<ItemNotification> notifications;
}

File: app/src/main/java/com/mmutert/freshfreezer/data/ItemAndNotifications.kt
package com.mmutert.freshfreezer.data

import androidx.room.Embedded
import androidx.room.Relation

class ItemAndNotifications(
    @Embedded
    val item: StorageItem,

    @Relation(parentColumn = "id", entityColumn = "item_id")
    val notifications: List<ItemNotification>
)